fix: update deployment and notification steps in 1gp-publish.yml

cmitzel-ncino opened the pull request:
This pull request refactors and streamlines the CI/CD workflow templates for package build and publish processes. The main changes focus on standardizing environment variables, simplifying job steps, improving maintainability, and enhancing clarity in the workflow logic. Several steps are replaced with reusable actions, outputs and inputs are clarified, and naming conventions are made consistent for better readability and future maintenance.

**Workflow simplification and maintainability:**

* Replaced custom script steps for extracting package info and path filtering with reusable actions (`getPackageInfo` and `pathFiltering`) in both `1gp-ci.yml` and `1gp-publish.yml`, reducing code duplication and improving maintainability. [[1]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL29-R77) [[2]](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eL1-L107)
* Consolidated and clarified environment variables and permissions, using consistent quoting and variable names across both workflow files. [[1]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL29-R77) [[2]](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eL1-L107)

**Job and step refactoring:**

* Renamed jobs and steps for clarity and consistency (e.g., `Sync-And-Transition-Tickets` instead of `swordfish`, `Package-Create` instead of `Publish-Package`, and more descriptive step names throughout). [[1]](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eL1-L107) [[2]](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eR86-R112)
* Improved output usage and passing between jobs, ensuring correct references and variable names (e.g., using `needs.Sync-And-Transition-Tickets.outputs.tag` instead of previous job names). [[1]](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eR86-R112) [[2]](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eR132-R155)

**Build and test improvements:**

* Updated CLI and tool setup steps to use workflow variables for versioning, and standardized installation flags (`ncino-cli-install`, `setup-salesforce-version`). [[1]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL180-L189) [[2]](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eR132-R155)
* Fixed conditional logic and step ordering to ensure that deployment and testing only proceed on successful org setup, and improved cleanup steps for reliability.

**Minor fixes and consistency:**

* Corrected output and environment variable references in echo and print steps, ensuring accurate logging and summary generation. [[1]](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eL120-R75) [[2]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbR153-R158)
* Removed redundant or obsolete comments and streamlined YAML formatting for easier future updates.

cmitzel-ncino committed:
fix: update deployment and notification steps in 1gp-publish.yml

- Replaced the deploy command with ncino/action-swordfish for better metadata deployment.
- Updated Slack notification actions to use version 2 of the action, enhancing message formatting.
- Adjusted conditions for failure notifications to check the outcome of the new deployment step.
- Changed the version reference for package uploads to use the output from the Sync-And-Transition-Tickets job.

cmitzel-ncino committed:
refactor: streamline CI workflows and introduce new actions for package info retrieval

- Removed redundant environment variables and simplified job steps in `1gp-ci.yml` and `1gp-publish.yml`.
- Introduced new composite actions for package info retrieval and path filtering to enhance modularity.
- Updated job dependencies and outputs to improve workflow clarity and efficiency.
- Enhanced Slack notification handling and adjusted conditions for build requirements.

abath-ncino commented on workflow-templates/1gp-publish.yml:
```diff
@@ -211,89 +167,75 @@ jobs:
+      - name: Running flow version notifications
+        run: |
+          pip install defusedxml
+          $DELIVERY_SCRIPTS_PATH/flow_version_notifications.py
```
hmm only 8 repos have this 🤔

cmitzel-ncino replied:
I can remove it, new repos shouldn't need this right?

abath-ncino replied:
there shouldn't be any new 1GP repos anyway

cmitzel-ncino committed:
chore: remove unnecessary flow version notifications step in 1gp-publish.yml

- Deleted the step for running flow version notifications to streamline the workflow.
- This change simplifies the CI process by eliminating redundant actions.
